.clang-format

```diff
@@ -0,0 +1,34 @@
+---
+BasedOnStyle: Google
+---
+Language: Cpp
+ColumnLimit: 100
+IndentWidth: 8
+ContinuationIndentWidth: 8
+ConstructorInitializerIndentWidth: 8
+TabWidth: 8
```

Comment:
Why 8? This will cause things to wrap very quickly, AFAICT.

Reply:
I don't have a strong preference, but I would argue we should apply the formatting to the default case, so that it is 100 column width on places like Github, and most unconfigured editors. It shouldn't affect CFEngine too much, because if you are submitting new code with tabs, the line width also increases, so it shouldn't break unless you are very deep.

What do other people think?

Reply:
it's `4` in [our golang settings](https://github.com/mendersoftware/mendertesting/blob/master/.golangci.yml#L48)
and thus I would vote for 4 (also considering that this codestyle would be a guideline for all our projects, that would be a good middleground I think)...

also -> https://github.com/golang/go/commit/014f3dcc837cb6789076cff4fccaa3bd221f823e <- go actually abandoned the config option to prevent non-uniform formatting - we deliberately ignored that in the golangci settings, so I think it's ok to ignore it here...

Reply:
> it's `4` in [our golang settings](https://github.com/mendersoftware/mendertesting/blob/master/.golangci.yml#L48)

Oh, I stand corrected. Then I guess there's no discussion, we've already been running with it for a long time.